webplc: simple web interface in golang

bnewbold opened the pull request:
This adds a simple HTML website with lookup interface, implemented in golang.

It also adds an openapi3 YAML spec file for the directory HTTP API, which is rendered using redoc in the website, but could be imported and reused in other ways.

The golang webapp is kind of overkill, copied from `bskyweb` (the `social-app` webserver, which mostly just returns static assets, but also does some social card HTML generation and API fetches). It would have been simpler to do static site generation and a bit of JS to do in-browser DID PLC fetching and rendering, but 1) i'm more familiar with golang and 2) have a side-quest goal to re-implement the core did-method-plc types and verification logic in golang, in this repo, and this seemed like a way to get that ball rolling. The motivation for 2) is to do it from the spec to ensure the spec is complete, and prove interoperability.

The website probably needs copy editing. I'm not really attached to the design/CSS, but think it just needs to be visually distinct from Bluesky branding and otherwise shouldn't spend much time on it.

There is an example deployment of this branch at: https://web.plc.staging.bsky.dev/

and in particular the openapi/redoc schema at: https://web.plc.staging.bsky.dev/api/redoc

bnewbold commented:
We were not sure if we wanted to launch this, but consensus from a week or two back is to go ahead with this public web site.

@emilyliu7321 can you do a review for copy on the front page? I expect that we'll do an iteration of this website in the next month or two, so splitting "small changes" (to make before merging this) and "large changes" (for follow-up) might make sense.

dholms reviewed (approved):
didn't really parse through all the code but:
- website looks good
- api docs look good

Small tweaks:
- we don't explain the 410 status code in the api docs
- maybe a quick note about go-didplc in the README? could be confusing to stumbled upon

Also: where are we planning to deploy this at? like `web.plc.directory`?

bnewbold commented:
updated README a bit, and fixed typos so that 410 shows up in specs.

planning to deploy to `web.plc.directory`. will update to redirect `plc.directory` with no path to that address in a follow-up.
